main.c

```diff
@@ -120,11 +120,14 @@ int main(int argc, char **argv) {
 		fprintf(stderr, "Unable to open %s\n", context.rom_file);
 		exit(1);
 	}
-
 	while (context.length--) {
 		fputc(0xff, context.rom);
 	}
-
+	if (context.input == NULL) {
+		fclose(context.rom);
+		return 0;
+	}
+	
 	input = context.input;
 	while (1) {
```

Comment:
The whole thing in while(input) ,but the rom file needs to be generated even the input== NULL.
Correct me if I understand it wrong .